src/with-theme.js

```diff
@@ -66,8 +91,11 @@ export default function withTheme(
         // as our `render` method without going through lifecycle hooks.
         // Also allows us to forward the context in the scenario where
         // a user wants to add more context.
-        // eslint-disable-next-line babel/new-cap
-        return ComponentToTheme({...this.props, ...this.state}, this.context)
+
+        /* eslint-disable max-len */
+        // $FlowFixMe: This may produce undesirable behavior when ComponentToTheme is a class instead of a function.
```

Comment:
Correct, but this is not technically a public API (at least, it wont be supported). It's only used for creating glamorous components. Didn't we discuss somewhere that we could make things work by extending `ThemedComponent` instead of doing this weird stuff though? I'll give it a look.